### PR TITLE
feat: provides the original item as onChange argument at Dropdown

### PR DESCRIPTION
**Pull request by gabrielmicko**
## PR checklist

- [x] I have followed the ["Committing and publishing"](https://github.com/quid/refraction/blob/master/CONTRIBUTING.md) guidelines;
- [x] I have added unit tests to cover my changes;
- [x] I updated the documentation and examples accordingly;

## Changes description
Here is a simple test example what capability are we adding: [line 14](https://github.com/quid/refraction/blob/a1b2df91d67d06c8dfc17ed686694d8e570d55c9/packages/react-dropdown/src/index.test.js#L114)
 

## Affected packages

- @quid/react-dropdown

### CODE REVIEW

File: packages/react-dropdown/src/Items.js [hunk 1]
```diff
@@ -100,8 +102,12 @@ export default function DropdownItems({
 }: Props) {
   return (
     <Items>
-      {items.map((item, index) => {
-        const itemIndex = item.hasOwnProperty('index') ? item.index : index;
+      {items.map((item: DropdownItemWithIndex, index: number) => {
+        let itemIndex = index;
+        if (item.hasOwnProperty('index')) {
+          itemIndex = item.index;
+          delete item.index;
```
Comment (FezVrasta) [inline packages/react-dropdown/src/Items.js]: I think the way you defined a new item object previously was easier to understand.

Comment (gabrielmicko) [reply packages/react-dropdown/src/Items.js]: I have to remove the index at some point. Can you propose what you mean exactly containing the delete part?

Comment (gabrielmicko) [reply packages/react-dropdown/src/Items.js]: do you mean this: 
```
const itemIndex = item.hasOwnProperty('index') ? item.index : index;
delete item.index;
```

Comment (FezVrasta) [reply packages/react-dropdown/src/Items.js]: I see that previously you created a new object to remove the index:

```
item: {
  id: item.id,
  label: item.label,
},
```

can't you do the same?

Comment (gabrielmicko) [reply packages/react-dropdown/src/Items.js]: I need to pass the same stuff that item has expect index. For example if user provides items like these:
```
const items = {
   id: 1,
   label: 'Hey',
   puppy: 'huge'
}
```

We have to make sure to pass it back the same way.

What I could do is:

```
item: {
  ...item
  id: item.id,
  label: item.label,
},
```
but I should remove `index`. I think. What do you think?

Comment (FezVrasta) [reply packages/react-dropdown/src/Items.js]: in that case I'd go with a spread operator:

```
const { index, ...itemWithoutIndex } = item;
```

**Pull request merged**